lib/matplotlib/markers.py

```diff
@@ -229,7 +230,9 @@ def set_fillstyle(self, fillstyle):
             raise ValueError("Unrecognized fillstyle %s"
                              % ' '.join(self.fillstyles))
         self._fillstyle = fillstyle
-        self._recache()
+        # do not call _recache() if marker is not yet set up
+        if self._marker is not None:
```

Comment:
I do not like this. Otherwise we can extract body of `set_fillstyle` without `_recache()` call to an internal method and call it from `__init__` and `set_fillstyle`.